src/core/zowe/core_for_zowe_sdk/profile_manager.py

```diff
@@ -58,260 +71,169 @@ def config_appname(self) -> str:
-
-        try:
-            base_profile_name = self.get_profilename_from_profiletype(
-                profile_jsonc=profile_jsonc, profile_type="base"
+        except ProfileNotFound:
+            warnings.warn(
+                f"Profile not found in file '{cfg.filename}', trying to return base profile instead.",
+                ProfileNotFoundWarning,
             )
-            base_props = profile_jsonc["profiles"][base_profile_name].get(
-                "properties", {}
+            try:
+                cfg_profile, cfg_profile_name = cfg.get_profile(
+                    profile_name=BASE_PROFILE, profile_type=BASE_PROFILE
+                )
+            except Exception as exc:
+                warnings.warn(
+                    f"Base Profile not found in file '{cfg.filename}' because {type(exc).__name__}'{exc}', "
+                    f"returning empty profile.",
+                    ProfileNotFoundWarning,
+                )
+        except SecureProfileLoadFailed:
+            warnings.warn(
+                f"Config '{cfg.filename}' has no saved secure properties.",
+                SecurePropsNotFoundWarning,
             )
-        except KeyError as exc:
-            raise ProfileNotFound(
-                profile_name=base_profile_name, error_msg=str(exc)
-            ) from exc
-
-        secure_props = self.load_credentials()
-
-        secure_fields: list = profile_jsonc["profiles"][base_profile_name].get(
-            "secure", []
-        )
-
-        # load properties with key as profile.{profile_name}.properties.{*}
-        for (key, value) in secure_props.items():
-            if re.match(
-                "profiles\\." + base_profile_name + "\\.properties\\.[a-z]+", key
-            ):
-                property_name = key.split(".")[3]
-                if property_name in secure_fields:
-                    base_props[property_name] = value
-                    secure_fields.remove(property_name)
-
-        if len(secure_fields) > 0:
-            raise SecureValuesNotFound(secure_fields)
-
-        return base_props
+        except SecurePropsNotFoundWarning:
+            if profile_name:
+                warnings.warn(
+                    f"Secure properties of profile '{profile_name}' from file '{cfg.filename}' were not found "
+                    f"hence profile not loaded.",
+                    SecurePropsNotFoundWarning,
+                )
+            else:
+                warnings.warn(
+                    f"Secure properties of profile type '{profile_type}' from file '{cfg.filename}' were not found "
+                    f"hence profile not loaded.",
+                    SecurePropsNotFoundWarning,
+                )
+        except Exception as exc:
+            warnings.warn(
+                f"Could not load {config_type} '{cfg.filename}' at '{cfg.filepath}'"
+                f"because {type(exc).__name__}'{exc}'.",
+                ConfigNotFoundWarning,
+            )
```

Comment:
I think we let `cfg.get_profile` raise the `ProfileNotFound` error.
I'm wondering if all `__init__`  in `exceptions.py` should have default values (in this case `profile_name` and `error_msg`) in order to print a better error message

current behavior
```
TypeError'ProfileNotFound.__init__() missing 1 required positional argument: 'error_msg''."
```

with `""` as defaults
```
ProfileNotFound: Failed to load profile Could not find profile as both profile_name and profile_type is not set.: 
```